### ISSUE TITLE
Version 1.4.1 returns different CPU recommendation than v 1.6.0

### ISSUE THREAD

**Issue by lukastopiarz** [robusta-dev/krr#157]
**Describe the bug**
Hi,
I recently tried to switch from KRR v1.4.1 to KRR v1.6.0 (I skipped v1.5.3 due to the mentioned computational bug) but found that it returns a different CPU recommendation. By 1 core per container, so it's kind of a huge difference.

**To Reproduce**
python3 krr.py simple --prometheus-url https://${PROMETHEUS_URL} --history_duration 720 --timeframe_duration 5 --memory_buffer_percentage 5 --formatter yaml -n ${NAMESPACE}

Abbreviated output from 1.4.1:
```bash
  recommended:
    requests:
      cpu:
        value: 3.664
        severity: WARNING
      memory:
        value: 10058989568.0
        severity: CRITICAL
```

Redacted output from 1.6.0:
```bash
  recommended:
    requests:
      cpu:
        value: 4.672
        severity: CRITICAL
      memory:
        value: 10058989568.0
        severity: CRITICAL
```

**Screenshots**
<img width="1586" alt="image" src="https://github.com/robusta-dev/krr/assets/15138454/0b48e9c3-4a2c-472a-b5f1-85ee9a567b3e">

**Additional context**
Which version returns more accurate data? Why is there so big a difference? In my case, if there are 20 PODs running in production it makes 72 vs 93 Cores recommended capacity allocation.

**Comment by LeaveMyYard** [comment]
Hey, @lukastopiarz!

Basically what we have done in 1.6 is that we moved all the formulas to prometheus itself, increasing default points resolution, what increased the accuracy.

So the short answer: 
1.6 should give more accurate calculations

Long answer:
with your case I think it actually matters to consider if the default logic suits you well. As I understand, your pods are giving large spikes for a lot of the time with big CPU usage. Depending on specific requirements of your app, you might consider playing with `--cpu_percentile`, for example to make it 95 or 90. This will decrease the resulted request, but will make some of the pods to utilize shared CPU (when using more then the request). This might cause throtling, if peaks are happening all together. But, depending on your exact app, this might be OK. Or, if you really want to be 100% sure that your pods are getting all the CPU they require - just make them as default 99p tells.
To sum up: balance between giving your pods all required resources vs optimizing costs is not trivial task and really depends on your exact requirements.

**Comment by lukastopiarz** [comment]
@LeaveMyYard thank you for providing me with the clarification
